Options `thresh_PC` and `width_alpha` are still not working properly

See discussions in #200 and #83.

Regarding `width_alpha`:

-  as far as I can tell, `width_alpha` is correctly used to initialize the prior of `width`, and in `Configuration.make_sigmoid`, from which is should be picked up everywhere else

-  the problem reported by @lschwetlick is easily reproducible. The following snippet fails for `width_alpha > 0.05` 

```
import numpy as np
import psignifit

data = np.array([[0.0010, 45.0000, 90.0000],
                 [0.0015, 50.0000, 90.0000],
                 [0.0020, 44.0000, 90.0000],
                 [0.0025, 44.0000, 90.0000],
                 [0.0030, 52.0000, 90.0000],
                 [0.0035, 53.0000, 90.0000],
                 [0.0040, 62.0000, 90.0000],
                 [0.0045, 64.0000, 90.0000],
                 [0.0050, 76.0000, 90.0000],
                 [0.0060, 79.0000, 90.0000],
                 [0.0070, 88.0000, 90.0000],
                 [0.0080, 90.0000, 90.0000],
                 [0.0100, 90.0000, 90.0000]])

res = psignifit.psignifit(data, width_alpha=0.1)
```

- the exception is coming from _check_prior, which controls that probabilities are never 0
```
File ~/o/psignifit/psignifit/_priors.py:107, in _check_prior(priors, name, values)
    105     assert np.all(np.isfinite(result)), f"Prior '{name}' returns non-finite values."
    106     assert np.all(result >= 0), f"Prior '{name}' returns negative values."
--> 107     assert np.all(result != 0), f"Prior '{name}' returns zeros."
[...]
AssertionError: Prior 'width' returns zeros.
```

- there is nothing wrong in principle with the prior being 0 somewhere, in fact the prior for width is defined so that "the range of the stimulus levels with cosine fall offs to 0 at the minimal difference of two stimulus levels and at 3 times the range of the tested stimulus levels. Also, whether or not the prior is zero depends on the range on which the prior is calculated, which is different between `_priors._check_prior` and  when the posterior is computed in `_parameter.parameter_bounds`

- I think the check is there because in `_posterior.log_posterior` we compute the logarithm of the prior directly, which gives `-inf`  when the prior is 0
```
    # add priors on the corresponding axis
    p += np.log(priors['threshold'](thres))
    p += np.log(priors['width'](width))
    p += np.log(priors['lambda'](lambd))
```


My recommendation:
1) in check_prior, change
```
 assert np.all(result != 0), f"Prior '{name}' returns zeros."
 ```
 
 to 
 
 ```
  assert np.any(result > 0), f"Prior '{name}' returns zeros."
  ```
  
with the intent of checking that at least one value is non-zero, but the rest can be zero without problems

2) use a robust way to compute the logarithm. in `log_posterior`. A common way to do is to clip the values to something small, e.g. in sklearn they do `y_pred = np.clip(y_pred, eps, 1 - eps)` where `eps = np.finfo(y_pred.dtype).eps`

--- a/docs/examples/advanced-options.md
+++ b/docs/examples/advanced-options.md
@@ -104,13 +104,9 @@ res = ps.psignifit(data, **options)
 
 ### Threshold definition (`thresh_PC`)
 
-```{warning}
-This option is only implemented in the MATLAB version of psignifit.
-For the python version the default value cannot be changed. 
-Adding this functionality is still work in progress.
-```
-
-This option sets the proportion correct to correspond to the threshold on the *unscaled* sigmoid. Possible values are in the range from 0 to 1, default is 0.5. The default corresponds to 75\% in a 2AFC task (midway between the guess rate of 50 % and ceiling performance 100%).
+This option sets the proportion correct to correspond to the threshold on the *unscaled* sigmoid. Possible values 
+are in the range from 0 to 1, default is 0.5. The default corresponds to 75\% in a 2AFC task (midway between the 
+guess rate of 50 % and ceiling performance 100%).
 
 To set it to a different value, for example to 90 %, you'll do
 

--- a/docs/how_to/How-to-Change-the-Threshold-Percent-Correct.md
+++ b/docs/how_to/How-to-Change-the-Threshold-Percent-Correct.md
@@ -13,13 +13,9 @@ kernelspec:
 
 # Change the threshold percentage correct 
 
-```{warning}
-This option is only implemented in the MATLAB version of psignifit.
-For the python version the default value cannot be changed. 
-Adding this functionality is still work in progress.
-```
-
-This option sets the proportion correct to correspond to the threshold on the *unscaled* sigmoid. Possible values are in the range from 0 to 1, default is 0.5. The default corresponds to 75\% in a 2AFC task (midway between the guess rate of 50 % and ceiling performance 100%).
+This option sets the proportion correct to correspond to the threshold on the *unscaled* sigmoid. Possible values are 
+in the range from 0 to 1, default is 0.5. The default corresponds to 75\% in a 2AFC task (midway between the guess 
+rate of 50 % and ceiling performance 100%).
 
 To set it to a different value, for example to 90 %, you'll do
 
@@ -28,3 +24,7 @@ options = {'thresh_PC': .9}
 ```
 
 Note that this corresponds to a 95 \% in a 2AFC task.
+
+Be aware that even though the interpretation of the threshold parameter changes with different values for `thresh_PC`,
+the prior over the threshold remains unchanged, which means that the prior over psychometric functions will be
+shifted. If this is not what you intended, please define a custom prior over the threshold.

--- a/docs/how_to/How-to-Change-the-Threshold-Percent-Correct.md
+++ b/docs/how_to/How-to-Change-the-Threshold-Percent-Correct.md
@@ -25,6 +25,5 @@ options = {'thresh_PC': .9}
 
 Note that this corresponds to a 95 \% in a 2AFC task.
 
-Be aware that even though the interpretation of the threshold parameter changes with different values for `thresh_PC`,
-the prior over the threshold remains unchanged, which means that the prior over psychometric functions will be
-shifted. If this is not what you intended, please define a custom prior over the threshold.
+Be aware that the default prior over threshold assumes that the experimental stimulus range covers the range where 
+the threshold likely falls. If this doesn't match your setup, you'll need a custom prior.

--- a/psignifit/_priors.py
+++ b/psignifit/_priors.py
@@ -1,6 +1,7 @@
 # -*- coding: utf-8 -*-
-from typing import Tuple, Dict
 from functools import partial
+from typing import Tuple, Dict
+import warnings
 
 import numpy as np
 import scipy.stats
@@ -79,7 +80,10 @@ def eta_prior(x, k):
 def default_prior(parameter: str, stimulus_range: Tuple[float, float], width_min: float,
                    width_alpha: float, beta: float, threshold_percent_correct: float = 0.5) -> Dict[str, Prior]:
     if not np.isclose(threshold_percent_correct, 0.5):
-        raise ValueError("Default prior 'threshold' expects thresh_PC=0.5, got {thresh_PC = }")
+        warnings.warn("The `thresh_PC` parameters is set to a value different than 0.5. Be aware that the "
+                      "default prior over threshold remains unchanged, which means that the prior over psychometric "
+                      "functions will be shifted. If this is not what you intended, please define a custom prior "
+                      "over the threshold.")
 
     if parameter == 'threshold':
         prior = partial(threshold_prior, stimulus_range=stimulus_range)

--- a/psignifit/_priors.py
+++ b/psignifit/_priors.py
@@ -80,10 +80,10 @@ def eta_prior(x, k):
 def default_prior(parameter: str, stimulus_range: Tuple[float, float], width_min: float,
                    width_alpha: float, beta: float, threshold_percent_correct: float = 0.5) -> Dict[str, Prior]:
     if not np.isclose(threshold_percent_correct, 0.5):
-        warnings.warn("The `thresh_PC` parameters is set to a value different than 0.5. Be aware that the "
-                      "default prior over threshold remains unchanged, which means that the prior over psychometric "
-                      "functions will be shifted. If this is not what you intended, please define a custom prior "
-                      "over the threshold.")
+        warnings.warn(f"The `thresh_PC` parameter is set to {threshold_percent_correct}, not the default 0.5. "
+                      f"Be aware that the default prior over threshold assumes that the experimental stimulus range "
+                      f"covers the range where the threshold likely falls. If this doesn't match your setup, you'll "
+                      f"need a custom prior. See the documentation for guidance.")
 
     if parameter == 'threshold':
         prior = partial(threshold_prior, stimulus_range=stimulus_range)

--- a/psignifit/psigniplot.py
+++ b/psignifit/psigniplot.py
@@ -62,6 +62,7 @@ def plot_psychometric_function(result: Result,  # noqa: C901, this function is t
     else:
         ymin = 0
 
+    # --- Plot experimental data as dots
     x_data = data[:, 0]
     if plot_data:
         y_data = data[:, 1] / data[:, 2]
@@ -70,6 +71,7 @@ def plot_psychometric_function(result: Result,  # noqa: C901, this function is t
         size = np.sqrt(data[:, 2])*(data_size*100)
         ax.scatter(x_data, y_data, s=size, color=data_color, marker='.', clip_on=False)
 
+    # --- Plot the point estimate of the psychometric function
     sigmoid = config.make_sigmoid()
     x = np.linspace(x_data.min(), x_data.max(), num=1000)
     x_low = np.linspace(x[0] - extrapolate_stimulus * (x[-1] - x[0]), x[0], num=100)
@@ -85,16 +87,23 @@ def plot_psychometric_function(result: Result,  # noqa: C901, this function is t
     ax.plot(x_low, y[:len(x_low)], '--', c=line_color, lw=line_width, clip_on=False)
     ax.plot(x_high, y[-len(x_high):], '--', c=line_color, lw=line_width, clip_on=False)
 
+    # --- Plot the parameters
     if plot_parameter:
+        PC = config.thresh_PC
+        rescaled_PC = params['gamma'] + (1 - params['lambda'] - params['gamma']) * PC
+
+        # -- Vertical line at the point estimate of the threshold
         x = [params['threshold'], params['threshold']]
-        y = [ymin, params['gamma'] + (1 - params['lambda'] - params['gamma']) * config.thresh_PC]
+        y = [ymin, rescaled_PC]
         ax.plot(x, y, '-', c=line_color)
 
+        # -- Horizontal lines at lambda and gamma
         ax.axhline(y=1 - params['lambda'], linestyle=':', color=line_color)
         ax.axhline(y=params['gamma'], linestyle=':', color=line_color)
 
+        # -- Confidence interval of the threshold
         CI_95 = result.confidence_intervals['threshold']['0.95']
-        y = np.array([params['gamma'] + .5 * (1 - params['lambda'] - params['gamma'])] * 2)
+        y = np.array([rescaled_PC, rescaled_PC])
         ax.plot(CI_95, y, c=line_color)
         ax.plot([CI_95[0]] * 2, y + [-.01, .01], c=line_color)
         ax.plot([CI_95[1]] * 2, y + [-.01, .01], c=line_color)
@@ -467,7 +476,7 @@ def plot_2D_margin(result: Result,
         ax: Axis object on which to plot. Default is None (new axes are created)
     Returns:
         Axis object on which the plot has been made
-"""
+    """
     if ax is None:
         ax = plt.gca()
     if result.debug=={}:

